File: Book/C/Fig_5.6_staticSchedule.c
#include <stdio.h>
#include <math.h>
#include <omp.h>

#define ITER 100000000   // use a smaller value if available memory is small

void main()
{
   int i;
   double A[ITER];
   for (i = 0; i < ITER; ++i)
      A[i] = 2.0*i;

   #pragma omp parallel 
   {
      int i; 
      int id = omp_get_thread_num();
      double tdata = omp_get_wtime();
 
      #pragma omp for schedule(static)
      for (i = 1; i < ITER; i++) // notice i starts from 1 since
                                 // the denominator below cannot be 0
         A[i] = A[i] * sqrt(i) / pow(sin(i), tan(i));
      tdata = omp_get_wtime() - tdata;

      if (id == 0) printf("Time spent is %f sec \n", tdata);
   }
}

File: Book/Fortran/Fig_5.6_staticSchedule.f90
program main
   use omp_lib
   implicit none

   integer, parameter :: ITER = 100000000   ! use a smaller value if available memory is small
   integer :: i, id
   real*8 :: A(iter)
   real*8 :: tdata 
   real :: x

   do i = 1, ITER
      A(i) = 2.0 * i
   enddo

   !$omp parallel private (id, tdata, x) 

      id = omp_get_thread_num()
      tdata = omp_get_wtime()
 
      !$omp do schedule(static)
      do i = 1, ITER
         x =  i * 1.0
         A(i) = A(i) * sqrt(x) / (sin(x) ** tan(x))
      enddo

      tdata = omp_get_wtime() - tdata

      if (id == 0) print *, "Time spent is ", tdata, " sec"

   !$omp end parallel
end program main
